File: Applications/CSharp/Pointers/MaxArea.cs
int MaxArea(int[] height)
{
    var l = 0;
    var r = height.Length - 1;
    var maxW = 0;

    while (l < r)
    {
        var hL = height[l];
        var hR = height[r];

        var ab = Math.Min(hL, hR) * (r - l);
        if (ab > maxW)
        {
            maxW = ab;
        }

        if (hL < hR)
        {
            l++;
        }
        else
            r--;
    }

    return maxW;
}

File: Applications/Python/Pointers/MaxArea.py
def maxArea(self, height: List[int]) -> int:
    maxW = 0
    l = 0
    r = len(height) - 1

    while l < r:
        hl = height[l]
        hr = height[r]
        w = min(hl, hr) * (r - l)
        if w > maxW:
            maxW = w

        if hl < hr:
            l += 1
        else:
            r -= 1
    
    return maxW